### ISSUE TITLE
Hotspot does not read zimit-generated files

### ISSUE THREAD

**Issue by Popolechien** [kiwix/kiwix-hotspot#608]
Created an image using installer that had the [GCFD zim](https://library.kiwix.org/edu.gcfglobal.org_en_all_2021-03/A/index.html) (zimfarm receipe [here](https://farm.openzim.org/recipes/edu.gcfglobal.org_en)) 

Installation ran fine and upon opening the zim I get the expected "switch to https" warning. But then past this step I only get the little circle going round and round indefinitely and there is no access to content.

**Comment by rgaudin** [comment]
Doesn't look like a hotspot issue but downloading the ZIM to make sure and maybe transfer.

**Comment by rgaudin** [comment]
Can't reproduce ; would you mind retesting but opening the developer tools console first (⌥ + ⌘ + I  on Firefox) and copy/screenshot its output ? So we can see where it's stuck at.

**Comment by Popolechien** [comment]
It actually works on Firefox, but not on Chrome.
Looking at the dev console I get this:
`load.js:56 Uncaught (in promise) DOMException: Failed to register a ServiceWorker for scope ('https://kiwix.hello.hotspot/edu.gcfglobal.org_en_all.en/A/') with script ('https://kiwix.hello.hotspot/edu.gcfglobal.org_en_all.en/A/sw.js?replayPrefix=&root=edugcfglobal.org_en_all.en'): An SSL certificate error occurred when fetching the script.`

**Comment by kelson42** [comment]
@rgaudin Does this bug is confirmed? I would be really surprised!

**Comment by rgaudin** [comment]
Is the symptom the same for all of them ?

**Comment by Popolechien** [comment]
Yup. Little round thing going to no end.

**Comment by rgaudin** [comment]
Ok, you tried different browsers ?

**Comment by Popolechien** [comment]
Ok, got it:

Chrome (macOS, all extensions deactivated): does not work
Firefox (macOS), Chrome (android): I get the below activation page

![Capture d’écran 2021-07-27 à 10 06 07](https://user-images.githubusercontent.com/21675885/127119561-f2afd5d0-4634-43dd-87c0-43af2745ad0a.png)

This being said, Chrome is the dominant browser so it might be worth checking if it is an issue with macOS or if it happens on Windows as well.

**Comment by rgaudin** [comment]
Thanks ; that's helpful. Pretty sure it happens on all platform ; your initial message make it seems like it.

**Comment by Popolechien** [comment]
Yep, can confirm that Opera on PC (windows 10) is fine, but not Chrome (Win10).

**Comment by kelson42** [comment]
@rgaudin Both of us have been quite weak on that one :( @Popolechien Sorry for this pretty stupid and high impacting problem.

**Comment by rgaudin** [comment]
> Chrome (macOS, all extensions deactivated): does not work

Works fine for me, and that does not surprise me as I did test it when implementing the feature (had to take a screencast of it!)

<img width="860" alt="Screen Shot 2021-09-30 at 13 46 28" src="https://user-images.githubusercontent.com/57929/135467539-58fe34c7-d95c-4209-b78f-f82abee6075c.png">

Are you sure you had the problem on macOS? Trying to get a Windows running to test on Windows now.

**Comment by rgaudin** [comment]
OK, so the problem is after this tutorial, which works on all browsers. It's just that once you've added the exception, you don't see it anymore obviously.

I can reproduce the issue: certificate exception is added but chrome refuses to register the service worker still.

**Comment by rgaudin** [comment]
OK, unfortunately, Chrome now requires the certificate to be installed on the system to work with Service Workers.
The process is thus more complicated with many steps and of course it's different on Windows and macOS.

What should we do?

**Comment by Popolechien** [comment]
well the simpler for the user the better, no matter the cost on our side.

**Comment by rgaudin** [comment]
There is no simple way for the user.

The current (other browsers though) solution requires a few clicks in the browser to add an exception for the domain.
What we are looking at now is a many steps, multi-software process to get to the same result:

1. Click on “Not secure” button next to address bar
1. Click on “Certificate (invalid)”
1. In the popup, go to Details
1. Click export to file
1. Select a location and name to export to and save
1. Now in explorer, go to that location and double-click certificate
1. Choose install certificate
1. Select “place certificate in following store”
1. Click Browse
1. Select “Trusted root certificates authorities” from the list of similarly named options
1. Confirm
1. Confirm in the warning dialog
1. Exit the certificate manager
1. In chrome, refresh the page you were on

Steps 7-13 are visible [in this tutorial](https://techcommunity.microsoft.com/t5/windows-server-essentials-and/installing-a-self-signed-certificate-as-a-trusted-root-ca-in/ba-p/396105)

Another alternative, is to ask user:
- To use another browser (Firefox)
- To disable security in chrome by launching it with `--ignore-certificate-errors`.

**Comment by kelson42** [comment]
Invalidated with Zimit2
